clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_pipeline.py

```diff
@@ -0,0 +1,271 @@
+        else:
+            self.patch_size = 50
+            self.stride_size = 50
+
+        # The reading node
+        # -------------------------
+        read_node = npe.Node(name="ReadingFiles",
+                             iterables=[
+                                 ('t1w', t1w_files),
+                             ],
+                             synchronize=True,
+                             interface=nutil.IdentityInterface(
+                                 fields=self.get_input_fields())
+                             )
+
+        self.connect([
+            (read_node, self.input_node, [('t1w', 't1w')]),
+        ])
+
+    def build_output_node(self):
+        """Build and connect an output node to the pipeline."""
+        import nipype.interfaces.utility as nutil
+        from nipype.interfaces.io import DataSink
+        import nipype.pipeline.engine as npe
+        from clinica.utils.nipype import (fix_join, container_from_filename)
+        from clinica.utils.filemanip import get_subject_id
+        from .deeplearning_prepare_data_utils import get_data_datasink
+
+        # Write node
+        # ----------------------
+        write_node = npe.Node(
+                name="WriteCaps",
+                interface=DataSink()
+                )
+        write_node.inputs.base_directory = self.caps_directory
+        write_node.inputs.parameterization = False
+
+        # Node
+        # ----------------------
+        get_ids = npe.Node(
+                interface=nutil.Function(
+                    input_names=['image_id'],
+                    output_names=['image_id_out', 'subst_ls'],
+                    function=get_data_datasink),
+                name="GetIDs")
+
+        # Get subject ID node
+        # ----------------------
+        image_id_node = npe.Node(
+                interface=nutil.Function(
+                    input_names=['bids_or_caps_file'],
+                    output_names=['image_id'],
+                    function=get_subject_id),
+                name='ImageID'
+                )
+
+        # Find container path from t1w filename
+        # ----------------------
+        container_path = npe.Node(
+                nutil.Function(
+                    input_names=['bids_or_caps_filename'],
+                    output_names=['container'],
+                    function=container_from_filename),
+                name='ContainerPath')
+
+        self.connect([
+            (self.input_node, image_id_node, [('t1w', 'bids_or_caps_file')]),
+            (self.input_node, container_path, [('t1w', 'bids_or_caps_filename')]),
+            # (image_id_node, write_node, [('image_id', '@image_id')]),
+            (image_id_node, get_ids, [('image_id', 'image_id')]),
+            # Connect to DataSink
+            (get_ids, write_node, [('image_id_out', '@image_id')]),
+            (get_ids, write_node, [('subst_ls', 'substitutions')])
+
+            ])
```

Comment:
Based on my [previous comment](https://github.com/aramis-lab/clinica/pull/92/files#r427867347), this modification:
```suggestion
    def build_output_node(self):
        """Build and connect an output node to the pipeline."""
        import nipype.interfaces.utility as nutil
        from nipype.interfaces.io import DataSink
        import nipype.pipeline.engine as npe
        from clinica.utils.nipype import (fix_join, container_from_filename)
        from clinica.utils.filemanip import get_subject_id

        # Write node
        # ----------------------
        write_node = npe.Node(
                name="WriteCaps",
                interface=DataSink()
                )
        write_node.inputs.base_directory = self.caps_directory
        write_node.inputs.parameterization = False

        # Get subject ID node
        # ----------------------
        image_id_node = npe.Node(
                interface=nutil.Function(
                    input_names=['bids_or_caps_file'],
                    output_names=['image_id'],
                    function=get_subject_id),
                name='ImageID'
                )

        # Find container path from t1w filename
        # ----------------------
        container_path = npe.Node(
                nutil.Function(
                    input_names=['bids_or_caps_filename'],
                    output_names=['container'],
                    function=container_from_filename),
                name='ContainerPath')

        self.connect([
            (self.input_node, image_id_node, [('t1w', 'bids_or_caps_file')]),
            (self.input_node, container_path, [('t1w', 'bids_or_caps_filename')]),
            # Connect to DataSink
            (image_id_node, write_node, [('image_id', '@image_id')]),
            ])
```
was tested successfully for `whole` case. I think this will be the same for `slice` and `patch`.